src/main/java/org/opentripplanner/model/plan/StreetLeg.java

```diff
@@ -244,6 +254,27 @@ public String toString() {
       .toString();
   }
 
+  public StreetLeg withAccessibilityScore(float accessibilityScore) {
+    var copy = new StreetLeg(
+      mode,
+      startTime,
+      endTime,
+      from,
+      to,
+      distanceMeters,
+      generalizedCost,
+      legGeometry,
+      legElevation,
+      walkSteps,
+      accessibilityScore
+    );
+    copy.setPathwayId(pathwayId);
+    copy.setWalkingBike(walkingBike);
+    copy.setRentedVehicle(rentedVehicle);
+    copy.setVehicleRentalNetwork(vehicleRentalNetwork);
+    return copy;
+  }
+
```

Comment:
A StreetLegBuilder would simplify the above code, this duplicate a lot of the logic in the `withTimeShift()`.

Also move this method above the `toSting()` to keep it together with the`withTimeShift` method.